### PR TITLE
Remove deprecated react code and start initial upgrades for dependencies

### PR DESCRIPTION
**Pull request by jmkulwik**

### CODE REVIEW

File: apps/src/applab/AppLabTooltipOverlay.jsx [hunk 1]
```diff
@@ -1,6 +1,9 @@
-/** @file App Lab-specific Tooltip Overlay */
 import $ from 'jquery';
-import React, {PropTypes} from 'react';
+
+/** @file App Lab-specific Tooltip Overlay */
```
Comment (Hamms) [inline apps/src/applab/AppLabTooltipOverlay.jsx:1]: this should still go at the top of the file

File: apps/src/applab/ImportScreensDialog.jsx [hunk 2]
```diff
@@ -1,5 +1,8 @@
+import React from 'react';
+
 /* eslint-disable react/no-danger */
```
Comment (Hamms) [inline apps/src/applab/ImportScreensDialog.jsx:1]: this should go at the top of the file, too. Looks like there's a handful of other places where the beginning lines in the file have been reorganized when they shouldn't have been.

Comment (jmkulwik) [reply apps/src/applab/ImportScreensDialog.jsx:1]: Ahh sorry. I have the fix for that in another commit. Let me push that.

**Comment by davidsbailey** [comment]
I am still seeing some imports before file-level comments in the following files:

* apps/src/code-studio/pd/workshop_dashboard/components/date_picker.jsx
* apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx
* apps/src/code-studio/pd/workshop_dashboard/components/workshop_table_loader.jsx
* apps/src/gamelab/AnimationTab/ItemLoopToggle.jsx
* apps/src/gamelab/AnimationTab/NewListItem.jsx
* apps/src/gamelab/GameLabView.jsx
* apps/src/lib/tools/jsdebugger/JsDebugger.jsx
* apps/src/storage/dataBrowser/AddKeyRow.jsx
* apps/src/storage/dataBrowser/ColumnHeader.jsx
* apps/src/storage/dataBrowser/ColumnMenu.jsx
* apps/src/storage/dataBrowser/DataOverview.jsx
* apps/src/storage/dataBrowser/DataProperties.jsx
* apps/src/storage/dataBrowser/DataTable.jsx
* apps/src/storage/dataBrowser/TableControls.jsx
* apps/src/templates/Button.jsx
* apps/src/templates/Certificate.jsx
* apps/src/templates/ResizablePanes.jsx
* apps/src/templates/instructions/InlineFeedback.jsx
* apps/src/templates/studioHomepages/UiTips.jsx
* apps/src/tutorialExplorer/tutorialExplorer.js

**Comment by davidsbailey** [comment]
just confirming that `import React, { Component } from 'react';` is still correct in react 16? in apps/src/code-studio/components/ShareDialog.jsx and apps/src/code-studio/components/ShareDisallowedDialog.jsx

**Review by davidsbailey — APPROVED** [review]
Looks good! Nice job leveraging an automated tool to get this syntax changed over!

**Comment by jmkulwik** [comment]
Responding to some comments earlier on this page:
import React, { Component } from 'react'; is still supported as far as I can tell
Thanks for finding the files with moved comments. I've fixed the last few.

**Review by Hamms — APPROVED** [review]

**Review by maddiedierker — APPROVED** [review]

**Comment by maddiedierker** [comment]
awesome clean-up! we should call this out to the group via email or slack to make sure nobody accidentally uses deprecated features like this and `React.createClass` while we're in limbo between React versions

**Pull request merged**